### PR TITLE
feat(Modelarts): add a new datasource to get resource flavors

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDatasourceResourceFlavors_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccDatasourceResourceFlavors_basic
=== PAUSE TestAccDatasourceResourceFlavors_basic
=== CONT  TestAccDatasourceResourceFlavors_basic
--- PASS: TestAccDatasourceResourceFlavors_basic (4.39s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	4.429s

```

### CODE REVIEW

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+		})
+	}
+	return rst
+}
+
+func flattenResourceFlavorsGpu(resp interface{}) []interface{} {
+	var rst []interface{}
+	curJson, err := jmespath.Search("spec.gpu", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing spec.gpu from response= %#v", resp)
```
Comment (houpeng80) [inline huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go]: should add the err info

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+		},
+	}
+	return rst
+}
+
+func flattenResourceFlavorsNpu(resp interface{}) []interface{} {
+	var rst []interface{}
+	curJson, err := jmespath.Search("spec.npu", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing spec.npu from response= %#v", resp)
```
Comment (houpeng80) [inline huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go]: ditto

File: docs/data-sources/modelarts_resource_flavors.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `tag` - (Optional, String) The tag key.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/modelarts_resource_flavors.md:25]: this is a little confusing, suggest to give an example value

File: docs/data-sources/modelarts_resource_flavors.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+# huaweicloud_modelarts_resource_flavors
+
+Use this data source to get resource flavors of ModelArts.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_modelart_resource_flavors" "test" {
+}
```
Comment (Jason-Zhang9309) [inline docs/data-sources/modelarts_resource_flavors.md]: if no parameters, this looks better
```suggestion
data "huaweicloud_modelart_resource_flavors" "test" {}
```

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go]: done

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_resource_flavors.go]: done

File: docs/data-sources/modelarts_resource_flavors.md (hunk 3 above)
Comment (hmbig2) [reply docs/data-sources/modelarts_resource_flavors.md:25]: ok, add parameters to the example

File: docs/data-sources/modelarts_resource_flavors.md (hunk 4 above)
Comment (hmbig2) [reply docs/data-sources/modelarts_resource_flavors.md]: ok,done

**Pull request merged**